### ISSUE TITLE
Validator Version 1.2.0 `TypeError: Cannot read properties of undefined (reading 'nodeInfo')`

### ISSUE THREAD

**Issue by theo-learner** [Shardeum/shardeum-bug-reporting#370]
##### What is the issue?

- CLI/GUI Running Version: 1.2.0 / 1.2.0
- Validator Running Version: 1.2.2

In the GUI, all menus go to a black screen and in the CLI, when I enter operator-cli status, I got a TypeError: Cannot read properties of undefined (reading 'nodeInfo') error. (RPC status 400: bad request)

discord: bhlee#7909
validator address: 0x706f9AC0C9d08d725B899cC8C2582620745CF930

#### What impact does the issue have?

All features such as node state checking/staking/unstaking are not available from the GUI. Crucially, the inability to check node state after a node run, even from the CLI, is fatal for node validators.

##### How to reproduce the issue? 

```
1. 
cd .shardeum
./shell.sh
operator-cli start
operator-cli status

2. connect <my_ip>:8080 
```

##### What other resources can you share regarding this issue?

error log:
```
 operator-cli status
AxiosError: Request failed with status code 404
    at settle (/home/node/app/cli/node_modules/axios/dist/node/axios.cjs:1859:12)
    at IncomingMessage.handleStreamEnd (/home/node/app/cli/node_modules/axios/dist/node/axios.cjs:2723:11)
    at IncomingMessage.emit (node:events:402:35)
    at endReadableNT (node:internal/streams/readable:1343:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  code: 'ERR_BAD_REQUEST',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [ 'xhr', 'http' ],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 2000,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function], Blob: null },
    validateStatus: [Function: validateStatus],
    headers: AxiosHeaders {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'axios/1.2.2',
      'Accept-Encoding': 'gzip, compress, deflate, br'
    },
    method: 'get',
    url: 'http://localhost:9001/nodeinfo',
    data: undefined
  },
  request: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)],
      prefinish: [Function: requestOnPrefinish]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: false,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    maxRequestsOnConnectionReached: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: false,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    _contentLength: 0,
    _hasBody: true,
    _trailer: '',
    finished: true,
    _headerSent: true,
    _closed: false,
    socket: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: 'localhost',
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      parser: null,
      _httpMessage: [Circular *1],
      timeout: 2000,
      [Symbol(async_id_symbol)]: 143,
      [Symbol(kHandle)]: [TCP],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: Timeout {
        _idleTimeout: 2000,
        _idlePrev: [TimersList],
        _idleNext: [TimersList],
        _idleStart: 3048,
        _onTimeout: [Function: bound ],
        _timerArgs: undefined,
        _repeat: null,
        _destroyed: false,
        [Symbol(refed)]: false,
        [Symbol(kHasPrimitive)]: false,
        [Symbol(asyncId)]: 151,
        [Symbol(triggerId)]: 146
      },
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(RequestTimeout)]: undefined
    },
    _header: 'GET /nodeinfo HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'User-Agent: axios/1.2.2\r\n' +
      'Accept-Encoding: gzip, compress, deflate, br\r\n' +
      'Host: localhost:9001\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: nop],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 80,
      protocol: 'http:',
      options: [Object: null prototype],
      requests: [Object: null prototype] {},
      sockets: [Object: null prototype],
      freeSockets: [Object: null prototype] {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 1,
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'GET',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    path: '/nodeinfo',
    _ended: true,
    res: IncomingMessage {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      socket: [Socket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      rawHeaders: [Array],
      rawTrailers: [],
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 404,
      statusMessage: 'Not Found',
      client: [Socket],
      _consuming: false,
      _dumped: false,
      req: [Circular *1],
      responseUrl: 'http://localhost:9001/nodeinfo',
      redirects: [],
      [Symbol(kCapture)]: false,
      [Symbol(kHeaders)]: [Object],
      [Symbol(kHeadersCount)]: 16,
      [Symbol(kTrailers)]: null,
      [Symbol(kTrailersCount)]: 0,
      [Symbol(RequestTimeout)]: undefined
    },
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'localhost',
    protocol: 'http:',
    _redirectable: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 0,
      _requestBodyBuffers: [],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [Circular *1],
      _currentUrl: 'http://localhost:9001/nodeinfo',
      _timeout: null,
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      'user-agent': [Array],
      'accept-encoding': [Array],
      host: [Array]
    }
  },
  response: {
    status: 404,
    statusText: 'Not Found',
    headers: AxiosHeaders {
      'x-powered-by': 'Express',
      'access-control-allow-origin': '*',
      'content-security-policy': "default-src 'none'",
      'x-content-type-options': 'nosniff',
      'content-type': 'text/html; charset=utf-8',
      'content-length': '147',
      date: 'Fri, 05 May 2023 13:21:46 GMT',
      connection: 'close'
    },
    config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 2000,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [AxiosHeaders],
      method: 'get',
      url: 'http://localhost:9001/nodeinfo',
      data: undefined
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      _closed: false,
      socket: [Socket],
      _header: 'GET /nodeinfo HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'User-Agent: axios/1.2.2\r\n' +
        'Accept-Encoding: gzip, compress, deflate, br\r\n' +
        'Host: localhost:9001\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/nodeinfo',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'localhost',
      protocol: 'http:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: '<!DOCTYPE html>\n' +
      '<html lang="en">\n' +
      '<head>\n' +
      '<meta charset="utf-8">\n' +
      '<title>Error</title>\n' +
      '</head>\n' +
      '<body>\n' +
      '<pre>Cannot GET /nodeinfo</pre>\n' +
      '</body>\n' +
      '</html>\n'
  }
}
/home/node/app/cli/build/src/utils/fetch-network-data.js:142
    if (nodeInfo === null || nodeInfo.nodeInfo === null) {
                                      ^

TypeError: Cannot read properties of undefined (reading 'nodeInfo')
    at fetchNodeInfo (/home/node/app/cli/build/src/utils/fetch-network-data.js:142:39)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/node/app/cli/build/src/node-commands.js:205:34
```

**Comment by MarcusWentz** [comment]
Related to:

https://github.com/Shardeum/shardeum-bug-reporting/issues/367

Did you configure your router to have TCP p2p ports 9001 and 10001 open?
Example:
https://www.att.com/support/article/u-verse-high-speed-internet/KM1010280/
Reference:
https://docs.shardeum.org/node/run/validator#step-2-download-and-install-validator

**Comment by theo-learner** [comment]
Network port is not an issue now.

**Comment by theo-learner** [comment]
I think we having similar issues with other object, not just `nodeInfo`. You need to suspect a bug at the code level.

**Comment by MarcusWentz** [comment]
A patch was added:
https://gitlab.com/shardeum/validator/cli/-/merge_requests/52

Do you still see the error when you uninstall then reinstall the validator?

**Comment by theo-learner** [comment]
![image](https://user-images.githubusercontent.com/9128515/236631823-510da5b6-79fa-4bae-a2db-ccc41ea04ba2.png)
1. Do you see any issues with my node?

2. Can the same version of Validator 1.2.2 have different source code behavior based on patch?

**Comment by MarcusWentz** [comment]
1. At a glance no, but maybe that patch was not added to your installation yet.

2. There can be a patch without a new release version yes. Please uninstall then reinstall the validator.

**Comment by theo-learner** [comment]
Thanks for your reply. I'm tried re-install the validator. now is 1.2.2.

another errror message after operation-cli status command is found: "Node not active in the network. Unable to fetch node-info". What should I do? @MarcusWentz

**Comment by swarna1101** [comment]
update to the latest node its solved in latest version 1.9.5.

**Comment by theo-learner** [comment]
> update to the latest node its solved in latest version 1.9.5.

Great. Let me check it. Thanks!
